
<!DOCTYPE html>

<html lang="python">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

    <title>Alignment &#8212; GenBioX 0.1 documentation</title>
    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="_static/alabaster.css" />
    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/_sphinx_javascript_frameworks_compat.js"></script>
    <script src="_static/doctools.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Annotation" href="annotation.html" />
    <link rel="prev" title="Statistical Analysis" href="statistical_analysis.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <section id="alignment">
<h1>Alignment<a class="headerlink" href="#alignment" title="Permalink to this heading">¶</a></h1>
<p>Perform sequence alignment to a reference genome or to other sequences in a database</p>
<section id="align">
<h2>align<a class="headerlink" href="#align" title="Permalink to this heading">¶</a></h2>
<p>Aligns a given sequence to a reference genome using a specified alignment algorithm.</p>
<p>Usage:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">GenBioX.alignment</span> <span class="k">as</span> <span class="nn">a</span>

<span class="n">a</span><span class="o">.</span><span class="n">align</span><span class="p">(</span><span class="n">seq</span><span class="p">,</span> <span class="n">ref</span><span class="p">,</span> <span class="n">algorithm</span><span class="o">=</span><span class="s1">&#39;global&#39;</span><span class="p">)</span>
</pre></div>
</div>
</section>
<section id="read-alignment">
<h2>read_alignment<a class="headerlink" href="#read-alignment" title="Permalink to this heading">¶</a></h2>
<p>Maps high-throughput sequencing reads to a reference genome to identify their genomic location and alignment quality.</p>
<p>Usage:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">a</span><span class="o">.</span><span class="n">read_alignment</span><span class="p">(</span><span class="n">reads</span><span class="p">,</span> <span class="n">reference_genome</span><span class="p">)</span>
</pre></div>
</div>
</section>
<section id="pairwise-alignment">
<h2>pairwise_alignment<a class="headerlink" href="#pairwise-alignment" title="Permalink to this heading">¶</a></h2>
<p>performs pairwise_alignment of two sequences</p>
<p>Usage:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">a</span><span class="o">.</span><span class="n">pairwise_alignment</span><span class="p">(</span><span class="s2">&quot;seq.fasta&quot;</span><span class="p">,</span> <span class="s2">&quot;seq2.fasta&quot;</span><span class="p">)</span>

<span class="n">a</span><span class="o">.</span><span class="n">pairwise_alignment</span><span class="p">(</span><span class="s2">&quot;seq.fasta&quot;</span><span class="p">,</span> <span class="s2">&quot;seq2.fasta&quot;</span><span class="p">,</span> <span class="n">gap_penalty</span><span class="o">=-</span><span class="mi">10</span><span class="p">,</span> <span class="n">match_score</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span> <span class="n">mismatch_penalty</span><span class="o">=-</span><span class="mi">4</span><span class="p">)</span>
</pre></div>
</div>
</section>
<section id="merge-alignment">
<h2>merge_alignment<a class="headerlink" href="#merge-alignment" title="Permalink to this heading">¶</a></h2>
<p>Merges two sequence alignments to produce a single alignment.</p>
<p>Usage:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">a</span><span class="o">.</span><span class="n">merge_alignments</span><span class="p">(</span><span class="n">aln1</span><span class="p">,</span> <span class="n">aln2</span><span class="p">)</span>
</pre></div>
</div>
</section>
<section id="evaluate-alignment-quality">
<h2>evaluate_alignment_quality<a class="headerlink" href="#evaluate-alignment-quality" title="Permalink to this heading">¶</a></h2>
<p>Evaluates the quality of a sequence alignment based on various metrics such as coverage, accuracy, and gap distribution.</p>
<p>Usage:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">aln_quality</span> <span class="o">=</span> <span class="n">a</span><span class="o">.</span><span class="n">evaluate_alignment_quality</span><span class="p">(</span><span class="n">alignment</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">aln_quality</span><span class="p">)</span>
</pre></div>
</div>
</section>
<section id="extract-conserved-regions">
<h2>extract_conserved_regions<a class="headerlink" href="#extract-conserved-regions" title="Permalink to this heading">¶</a></h2>
<p>Extracts conserved regions from a multiple sequence alignment.</p>
<p>Usage:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">conserved_regions</span> <span class="o">=</span> <span class="n">a</span><span class="o">.</span><span class="n">extract_conserved_regions</span><span class="p">(</span><span class="n">alignment</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">conserved_regions</span><span class="p">)</span>
</pre></div>
</div>
</section>
</section>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="index.html">GenBioX</a></h1>








<h3>Navigation</h3>
<p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="data_preprocessing.html">Data Preprocessing</a></li>
<li class="toctree-l1"><a class="reference internal" href="statistical_analysis.html">Statistical Analysis</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Alignment</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#align">align</a></li>
<li class="toctree-l2"><a class="reference internal" href="#read-alignment">read_alignment</a></li>
<li class="toctree-l2"><a class="reference internal" href="#pairwise-alignment">pairwise_alignment</a></li>
<li class="toctree-l2"><a class="reference internal" href="#merge-alignment">merge_alignment</a></li>
<li class="toctree-l2"><a class="reference internal" href="#evaluate-alignment-quality">evaluate_alignment_quality</a></li>
<li class="toctree-l2"><a class="reference internal" href="#extract-conserved-regions">extract_conserved_regions</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="annotation.html">Annotation</a></li>
<li class="toctree-l1"><a class="reference internal" href="variant_analysis.html">variant analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="gene_expression.html">gene expression</a></li>
<li class="toctree-l1"><a class="reference internal" href="comparative_genomics.html">comparative genomics</a></li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
      <li>Previous: <a href="statistical_analysis.html" title="previous chapter">Statistical Analysis</a></li>
      <li>Next: <a href="annotation.html" title="next chapter">Annotation</a></li>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>document.getElementById('searchbox').style.display = "block"</script>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2023, Sayali Garud.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 5.0.2</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.12</a>
      
      |
      <a href="_sources/alignment.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>