Search.setIndex({"docnames": ["alignment", "annotation", "comparative_genomics", "data_preprocessing", "gene_expression", "index", "statistical_analysis", "variant_analysis"], "filenames": ["alignment.rst", "annotation.rst", "comparative_genomics.rst", "data_preprocessing.rst", "gene_expression.rst", "index.rst", "statistical_analysis.rst", "variant_analysis.rst"], "titles": ["Alignment", "Annotation", "comparative genomics", "Data Preprocessing", "gene expression", "Welcome to GenBioX!", "Statistical Analysis", "variant analysis"], "terms": {"perform": [0, 2, 3, 4, 5], "sequenc": [0, 2, 3, 5, 6, 7], "refer": 0, "genom": [0, 3, 5], "other": 0, "databas": [0, 3, 7], "given": 0, "us": [0, 3, 4, 5, 6], "specifi": 0, "algorithm": 0, "usag": [0, 1, 2, 3, 4, 6, 7], "import": [0, 2, 5, 7], "genbiox": [0, 2, 7], "seq": [0, 3, 5, 6], "ref": 0, "global": [0, 2], "map": 0, "high": 0, "throughput": 0, "read": [0, 3, 4], "identifi": [0, 2, 3, 4, 7], "locat": [0, 1, 7], "qualiti": [0, 3, 5], "reference_genom": [0, 7], "two": [0, 2, 4], "fasta": [0, 3], "seq2": [0, 2], "gap_penalti": 0, "10": 0, "match_scor": 0, "5": [0, 3], "mismatch_penalti": 0, "4": 0, "merg": 0, "produc": 0, "singl": 0, "aln1": 0, "aln2": 0, "evalu": 0, "base": [0, 3], "variou": [0, 4, 5], "metric": 0, "coverag": 0, "accuraci": 0, "gap": 0, "distribut": [0, 3], "aln_qual": 0, "print": [0, 1, 3, 4, 5, 6], "extract": [0, 1, 5], "conserv": [0, 2], "region": [0, 2], "from": [0, 1, 3, 5, 7], "multipl": 0, "conserved_region": 0, "filter": [1, 5], "search": [1, 5], "genbank": [1, 5], "file": [1, 3, 4, 5], "an": [1, 5], "ann": 1, "gb": 1, "fiter_dict": 1, "product": 1, "hypothet": 1, "protein": [1, 6], "fa": 1, "queri": 1, "aminoglycosid": 1, "sa": [1, 5, 6], "gene": [1, 5, 7], "name": 1, "panda": 1, "datafram": [1, 3, 4], "loc": 1, "test": 1, "across": 2, "differ": [2, 5], "speci": 2, "strain": 2, "size": 2, "report": 2, "comparative_genom": [2, 5], "ca": [2, 5], "genome1": 2, "between": [2, 4], "more": [2, 4], "calcul": [2, 3, 4, 6], "pairwis": 2, "ident": 2, "seq1": 2, "method": [2, 4], "phylogenet": 2, "distanc": 2, "genome2": 2, "The": [3, 5, 6], "data_preprocess": [3, 5], "modul": [3, 5, 6], "provid": [3, 5, 6], "function": [3, 4, 5, 6, 7], "load": [3, 5], "control": [3, 5], "return": 3, "dictionari": 3, "contain": 3, "id": 3, "kei": 3, "correspond": 3, "valu": 3, "dp": [3, 5], "sequencing_data": 3, "retriev": 3, "public": 3, "ncbi": 3, "its": 3, "access": 3, "number": [3, 6], "accession_numb": 3, "vcf": 3, "variant": [3, 5], "vcf_path": 3, "check": 3, "qc": 3, "remov": [3, 4], "low": 3, "overal": [3, 5], "score": 3, "too": 3, "mani": 3, "filtered_data": 3, "min_avg_scor": 3, "20": 3, "max_low_quality_bas": 3, "each": 3, "typic": 3, "repres": 3, "phred": 3, "dentifi": 3, "adapt": 3, "mai": 3, "have": 3, "been": 3, "introduc": 3, "dure": 3, "librari": 3, "prepar": 3, "agatcggaagagcacacgtctgaactccagtcac": 3, "duplic": 3, "pcr": 3, "amplif": 3, "match": 3, "known": 3, "contamin": 3, "those": 3, "bacteri": 3, "viral": 3, "gener": [3, 6], "plot": 3, "summari": 3, "statist": [3, 5], "assess": 3, "per": 3, "length": [3, 4], "analysi": [4, 5], "convert": 4, "further": 4, "thi": [4, 5], "normal": 4, "raw": 4, "data": [4, 5, 7], "technic": 4, "variat": [4, 7], "bias": 4, "which": 4, "can": [4, 5], "affect": 4, "downstream": 4, "ge": [4, 5], "expression_data": 4, "quantil": 4, "get": 4, "group": 4, "label": 4, "For": [4, 5], "group_label": 4, "norm": 4, "differenti": [4, 5], "ar": 4, "condit": 4, "quantifi": 4, "level": 4, "rpkm": 4, "tpm": 4, "fpkm": 4, "input": 4, "counts_matrix": 4, "correl": 4, "phenotyp": [4, 7], "clinic": 4, "variabl": 4, "diseas": 4, "statu": 4, "ag": 4, "gender": 4, "help": 4, "potenti": 4, "biomark": 4, "therapeut": 4, "target": 4, "clinical_data": 4, "A": [5, 6], "comprehens": 5, "bioinformat": 5, "packag": 5, "purpos": 5, "set": 5, "tool": 5, "wide": 5, "rang": 5, "task": 5, "includ": 5, "preprocess": 5, "align": 5, "It": 5, "also": 5, "annot": [5, 7], "predict": [5, 7], "effect": [5, 7], "genet": [5, 6, 7], "analys": 5, "express": 5, "compar": 5, "visualis": 5, "interpret": 5, "result": 5, "research": 5, "To": 5, "you": 5, "pip": 5, "statistical_analysi": [5, 6], "variant_analysi": [5, 7], "va": [5, 7], "gene_express": 5, "call": 5, "ani": 5, "exampl": 5, "find": 5, "revers": [5, 6], "complement": [5, 6], "dna": 5, "atcgatcg": 5, "rev_comp": [5, 6], "reverse_compl": 5, "give": 5, "output": 5, "cgatcgat": 5, "read_fasta": 5, "fetch_seq": 5, "read_vcf": 5, "fasta_quality_check": 5, "filter_low_quality_read": 5, "quality_scor": 5, "trim_adapt": 5, "remove_dupl": 5, "filter_contamin": 5, "visualise_quality_metr": 5, "calculate_gc_cont": 5, "count_nucleotid": 5, "translat": 5, "read_align": 5, "pairwise_align": 5, "merge_align": 5, "evaluate_alignment_qu": 5, "extract_conserved_region": 5, "extract_annot": 5, "filter_annot": 5, "search_annot": 5, "extract_loc": 5, "visualize_annot": 5, "variant_cal": 5, "variant_annot": 5, "variant_effect_predict": 5, "splice_site_predict": 5, "phenotype_predict": 5, "gene_expression_norm": 5, "get_group_label": 5, "differential_gene_expression_analysi": 5, "gene_expression_quantif": 5, "gene_expression_correl": 5, "compare_genome_s": 5, "identify_conserved_region": 5, "pairwise_ident": 5, "phylogenetic_dist": 5, "index": 5, "page": 5, "gc": 6, "content": 6, "nucleotid": 6, "gc_count": 6, "count": 6, "c": 6, "g": 6, "t": 6, "u": 6, "standard": 6, "code": 6, "snp": 7, "indel": 7, "structur": 7, "ariant_cal": 7, "input_bam": 7, "output_vcf": 7, "inform": 7, "type": 7, "allel": 7, "frequenc": 7, "input_vcf": 7, "output_annovar_prefix": 7, "consequ": 7, "splice": 7, "analyz": 7, "site": 7, "motif": 7, "mouseretina": 4, "txt": 4, "read_gene_expression_fil": 5}, "objects": {}, "objtypes": {}, "objnames": {}, "titleterms": {"align": 0, "read_align": 0, "pairwise_align": 0, "merge_align": 0, "evaluate_alignment_qu": 0, "extract_conserved_region": 0, "annot": 1, "extract_annot": 1, "filter_annot": 1, "search_annot": 1, "extract_loc": 1, "visualize_annot": 1, "compar": 2, "genom": 2, "compare_genome_s": 2, "identify_conserved_region": 2, "pairwise_ident": 2, "phylogenetic_dist": 2, "data": 3, "preprocess": 3, "read_fasta": 3, "fetch_seq": 3, "read_vcf": 3, "fasta_quality_check": 3, "filter_low_quality_read": 3, "quality_scor": 3, "trim_adapt": 3, "remove_dupl": 3, "filter_contamin": 3, "visualise_quality_metr": 3, "gene": 4, "express": 4, "gene_expression_norm": 4, "get_group_label": 4, "differential_gene_expression_analysi": 4, "gene_expression_quantif": 4, "gene_expression_correl": 4, "welcom": 5, "genbiox": 5, "instal": 5, "get": 5, "start": 5, "content": 5, "indic": 5, "tabl": 5, "statist": 6, "analysi": [6, 7], "calculate_gc_cont": 6, "count_nucleotid": 6, "reverse_compl": 6, "translat": 6, "variant": 7, "variant_cal": 7, "variant_annot": 7, "variant_effect_predict": 7, "splice_site_predict": 7, "phenotype_predict": 7, "read_gene_expression_fil": 4}, "envversion": {"sphinx.domains.c": 2, "sphinx.domains.changeset": 1, "sphinx.domains.citation": 1, "sphinx.domains.cpp": 6, "sphinx.domains.index": 1, "sphinx.domains.javascript": 2, "sphinx.domains.math": 2, "sphinx.domains.python": 3, "sphinx.domains.rst": 2, "sphinx.domains.std": 2, "sphinx": 56}})