
<!DOCTYPE html>

<html lang="python">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

    <title>Data Preprocessing &#8212; GenBioX 0.1 documentation</title>
    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="_static/alabaster.css" />
    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/_sphinx_javascript_frameworks_compat.js"></script>
    <script src="_static/doctools.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Statistical Analysis" href="statistical_analysis.html" />
    <link rel="prev" title="Welcome to GenBioX&#39;s documentation!" href="index.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <section id="data-preprocessing">
<h1>Data Preprocessing<a class="headerlink" href="#data-preprocessing" title="Permalink to this heading">¶</a></h1>
<p>The data_preprocessing module provides functions for Loading sequence files, Quality control &amp; Preprocessing of the sequences from sequencing data.</p>
<section id="read-fasta">
<h2>read_fasta<a class="headerlink" href="#read-fasta" title="Permalink to this heading">¶</a></h2>
<p>Reads a FASTA file and returns a dictionary containing the sequence IDs as keys and the corresponding sequences as values.</p>
<p>Usage:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">GenBioX.data_preprocessing</span> <span class="k">as</span> <span class="nn">dp</span>

<span class="c1"># Load sequencing data</span>
<span class="n">seq</span> <span class="o">=</span> <span class="n">dp</span><span class="o">.</span><span class="n">read_fasta</span><span class="p">(</span><span class="s2">&quot;sequencing_data.fasta&quot;</span><span class="p">)</span>

<span class="c1"># Print the sequence</span>
<span class="nb">print</span><span class="p">(</span><span class="n">seq</span><span class="p">)</span>
</pre></div>
</div>
</section>
<section id="fetch-seq">
<h2>fetch_seq<a class="headerlink" href="#fetch-seq" title="Permalink to this heading">¶</a></h2>
<p>Retrieves a sequence from a public sequence database (such as NCBI) using its accession number.</p>
<p>Usage:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="c1"># Load sequencing data</span>
<span class="n">seq</span> <span class="o">=</span> <span class="n">dp</span><span class="o">.</span><span class="n">fetch_seq</span><span class="p">(</span><span class="n">database</span><span class="p">,</span> <span class="n">accession_number</span><span class="p">)</span>

<span class="c1"># Print the sequence</span>
<span class="nb">print</span><span class="p">(</span><span class="n">seq</span><span class="p">)</span>
</pre></div>
</div>
</section>
<section id="read-vcf">
<h2>read_vcf<a class="headerlink" href="#read-vcf" title="Permalink to this heading">¶</a></h2>
<p>Reads vcf file and returns a DataFrame with the variant data</p>
<p>Usage:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="c1"># Load data</span>
<span class="n">data</span> <span class="o">=</span> <span class="n">dp</span><span class="o">.</span><span class="n">read_vcf</span><span class="p">(</span><span class="n">vcf_path</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">data</span><span class="p">)</span>
</pre></div>
</div>
</section>
<section id="fasta-quality-check">
<h2>fasta_quality_check<a class="headerlink" href="#fasta-quality-check" title="Permalink to this heading">¶</a></h2>
<p>performs quality check on a sequence</p>
<p>Usage:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="c1"># Load data</span>
<span class="n">data</span> <span class="o">=</span> <span class="n">dp</span><span class="o">.</span><span class="n">fasta_quality_check</span><span class="p">(</span><span class="n">sequences</span><span class="p">,</span> <span class="n">min_length</span><span class="o">=</span><span class="mi">50</span><span class="p">,</span> <span class="n">max_length</span><span class="o">=</span><span class="mi">1000000</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">data</span><span class="p">)</span>
</pre></div>
</div>
</section>
<section id="filter-low-quality-reads">
<h2>filter_low_quality_reads<a class="headerlink" href="#filter-low-quality-reads" title="Permalink to this heading">¶</a></h2>
<p>Remove reads with low overall quality scores or with too many low-quality bases.</p>
<p>Usage:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="c1"># Load data</span>
<span class="n">data</span> <span class="o">=</span> <span class="n">dp</span><span class="o">.</span><span class="n">filter_low_quality_reads</span><span class="p">(</span><span class="n">quality_scores</span><span class="p">,</span> <span class="n">min_avg_score</span><span class="o">=</span><span class="mi">20</span><span class="p">,</span> <span class="n">max_low_quality_bases</span><span class="o">=</span><span class="mi">5</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">data</span><span class="p">)</span>
</pre></div>
</div>
</section>
<section id="quality-scores">
<h2>quality_scores<a class="headerlink" href="#quality-scores" title="Permalink to this heading">¶</a></h2>
<p>Calculate the quality scores for each base in a sequencing read, typically represented as a Phred score.</p>
<p>Usage:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="c1"># Load data</span>
<span class="n">data</span> <span class="o">=</span> <span class="n">dp</span><span class="o">.</span><span class="n">quality_scores</span><span class="p">(</span><span class="n">seq</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">data</span><span class="p">)</span>
</pre></div>
</div>
</section>
<section id="trim-adapters">
<h2>trim_adapters<a class="headerlink" href="#trim-adapters" title="Permalink to this heading">¶</a></h2>
<p>dentify and remove adapter sequences that may have been introduced during library preparation.</p>
<p>Usage:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="c1"># Load data</span>
<span class="n">data</span> <span class="o">=</span> <span class="n">dp</span><span class="o">.</span><span class="n">trim_adapters</span><span class="p">(</span><span class="n">sequence</span><span class="p">,</span> <span class="n">adapter</span><span class="o">=</span><span class="s1">&#39;AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC&#39;</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">data</span><span class="p">)</span>
</pre></div>
</div>
</section>
<section id="remove-duplicates">
<h2>remove_duplicates<a class="headerlink" href="#remove-duplicates" title="Permalink to this heading">¶</a></h2>
<p>Identify and remove duplicate reads that may have been introduced during PCR amplification.</p>
<p>Usage:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="c1"># Load data</span>
<span class="n">data</span> <span class="o">=</span> <span class="n">dp</span><span class="o">.</span><span class="n">remove_duplicates</span><span class="p">(</span><span class="n">seq</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">data</span><span class="p">)</span>
</pre></div>
</div>
</section>
<section id="filter-contaminants">
<h2>filter_contaminants<a class="headerlink" href="#filter-contaminants" title="Permalink to this heading">¶</a></h2>
<p>Identify and remove reads that match known contaminant sequences, such as those from bacterial or viral genomes.</p>
<p>Usage:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="c1"># Load data</span>
<span class="n">data</span> <span class="o">=</span> <span class="n">dp</span><span class="o">.</span><span class="n">filter_contaminants</span><span class="p">(</span><span class="n">seq</span><span class="p">,</span> <span class="n">contaminants</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">data</span><span class="p">)</span>
</pre></div>
</div>
</section>
<section id="visualise-quality-metrics">
<h2>visualise_quality_metrics<a class="headerlink" href="#visualise-quality-metrics" title="Permalink to this heading">¶</a></h2>
<p>Generate plots and summary statistics to assess the quality of sequencing data, such as per-base quality scores and read length distributions.</p>
<p>Usage:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="c1"># Load data</span>
<span class="n">data</span> <span class="o">=</span> <span class="n">dp</span><span class="o">.</span><span class="n">visualise_quality_metrics</span><span class="p">(</span><span class="n">sequences</span><span class="p">,</span> <span class="n">quality_scores</span><span class="p">)</span>
</pre></div>
</div>
</section>
</section>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="index.html">GenBioX</a></h1>








<h3>Navigation</h3>
<p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">Data Preprocessing</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#read-fasta">read_fasta</a></li>
<li class="toctree-l2"><a class="reference internal" href="#fetch-seq">fetch_seq</a></li>
<li class="toctree-l2"><a class="reference internal" href="#read-vcf">read_vcf</a></li>
<li class="toctree-l2"><a class="reference internal" href="#fasta-quality-check">fasta_quality_check</a></li>
<li class="toctree-l2"><a class="reference internal" href="#filter-low-quality-reads">filter_low_quality_reads</a></li>
<li class="toctree-l2"><a class="reference internal" href="#quality-scores">quality_scores</a></li>
<li class="toctree-l2"><a class="reference internal" href="#trim-adapters">trim_adapters</a></li>
<li class="toctree-l2"><a class="reference internal" href="#remove-duplicates">remove_duplicates</a></li>
<li class="toctree-l2"><a class="reference internal" href="#filter-contaminants">filter_contaminants</a></li>
<li class="toctree-l2"><a class="reference internal" href="#visualise-quality-metrics">visualise_quality_metrics</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="statistical_analysis.html">Statistical Analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="alignment.html">Alignment</a></li>
<li class="toctree-l1"><a class="reference internal" href="annotation.html">Annotation</a></li>
<li class="toctree-l1"><a class="reference internal" href="variant_analysis.html">variant analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="gene_expression.html">gene expression</a></li>
<li class="toctree-l1"><a class="reference internal" href="comparative_genomics.html">comparative genomics</a></li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
      <li>Previous: <a href="index.html" title="previous chapter">Welcome to GenBioX's documentation!</a></li>
      <li>Next: <a href="statistical_analysis.html" title="next chapter">Statistical Analysis</a></li>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>document.getElementById('searchbox').style.display = "block"</script>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2023, Sayali Garud.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 5.0.2</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.12</a>
      
      |
      <a href="_sources/data_preprocessing.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>