<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Welcome to GenBioX! &mdash; GenBioX 0.3.1 documentation</title>
      <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="_static/jquery.js"></script>
        <script src="_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/doctools.js"></script>
        <script src="_static/sphinx_highlight.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Data Preprocessing" href="data_preprocessing.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="#" class="icon icon-home">
            GenBioX
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="data_preprocessing.html">Data Preprocessing</a></li>
<li class="toctree-l1"><a class="reference internal" href="statistical_analysis.html">Statistical Analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="alignment.html">Alignment</a></li>
<li class="toctree-l1"><a class="reference internal" href="annotation.html">Annotation</a></li>
<li class="toctree-l1"><a class="reference internal" href="variant_analysis.html">variant analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="gene_expression.html">gene expression</a></li>
<li class="toctree-l1"><a class="reference internal" href="comparative_genomics.html">comparative genomics</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="#">GenBioX</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="#" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Welcome to GenBioX!</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/index.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <a class="logo-img reference internal image-reference" href="_images/logo.png"><img alt="_images/logo.png" class="logo-img align-center" src="_images/logo.png" style="width: 400px; height: 170px;" /></a>
<div style="height: 50px;"></div><section id="welcome-to-genbiox">
<h1>Welcome to GenBioX!<a class="headerlink" href="#welcome-to-genbiox" title="Permalink to this heading"></a></h1>
<p>A Comprehensive Bioinformatics package for genome analysis.</p>
<p>The purpose of this package is to provide a set of tools and functionalities to perform a wide range of genome analysis tasks, including loading, quality control, preprocessing, and alignment of sequencing data.
It also includes tools to extract and filter annotations from GenBank files, perform sequence annotation, predict the functional effects of genetic variants, analyse gene expression and differential gene expression, and perform comparative genomics.
The package provides visualisation tools to interpret the results of the analysis.
Overall, this package can be used for various genome analysis tasks in bioinformatics and genetics research.</p>
<section id="installation">
<h2>Installation<a class="headerlink" href="#installation" title="Permalink to this heading"></a></h2>
<p>To install GenBioX, you can use pip:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pip<span class="w"> </span>install<span class="w"> </span>GenBioX
</pre></div>
</div>
</section>
<section id="getting-started">
<h2>Getting started<a class="headerlink" href="#getting-started" title="Permalink to this heading"></a></h2>
<p>To use GenBioX, start by importing it:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">GenBioX</span>
</pre></div>
</div>
<p>To use Functions, start by importing different modules:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">GenBioX.data_preprocessing</span> <span class="k">as</span> <span class="nn">dp</span>
<span class="kn">import</span> <span class="nn">GenBioX.statistical_analysis</span> <span class="k">as</span> <span class="nn">sa</span>
<span class="kn">import</span> <span class="nn">GenBioX.alignment</span> <span class="k">as</span> <span class="nn">a</span>
<span class="kn">import</span> <span class="nn">GenBioX.annotation</span> <span class="k">as</span> <span class="nn">an</span>
<span class="kn">import</span> <span class="nn">GenBioX.variant_analysis</span> <span class="k">as</span> <span class="nn">va</span>
<span class="kn">import</span> <span class="nn">GenBioX.gene_expression</span> <span class="k">as</span> <span class="nn">ge</span>
<span class="kn">import</span> <span class="nn">GenBioX.comparative_genomics</span> <span class="k">as</span> <span class="nn">ca</span>
</pre></div>
</div>
<p>You can then call any of the functions provided by the package.
For example, to find the reverse complement of a DNA sequence:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">seq</span> <span class="o">=</span> <span class="s1">&#39;ATCGATCG&#39;</span>
<span class="n">rev_comp</span> <span class="o">=</span> <span class="n">sa</span><span class="o">.</span><span class="n">reverse_complement</span><span class="p">(</span><span class="n">seq</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">rev_comp</span><span class="p">)</span>
</pre></div>
</div>
<p>This will give output:</p>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>CGATCGAT
</pre></div>
</div>
<div class="toctree-wrapper compound">
<p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="data_preprocessing.html">Data Preprocessing</a><ul>
<li class="toctree-l2"><a class="reference internal" href="data_preprocessing.html#read-fasta">read_fasta</a></li>
<li class="toctree-l2"><a class="reference internal" href="data_preprocessing.html#fetch-seq">fetch_seq</a></li>
<li class="toctree-l2"><a class="reference internal" href="data_preprocessing.html#read-vcf">read_vcf</a></li>
<li class="toctree-l2"><a class="reference internal" href="data_preprocessing.html#fasta-quality-check">fasta_quality_check</a></li>
<li class="toctree-l2"><a class="reference internal" href="data_preprocessing.html#filter-reads">filter_reads</a></li>
<li class="toctree-l2"><a class="reference internal" href="data_preprocessing.html#quality-scores">quality_scores</a></li>
<li class="toctree-l2"><a class="reference internal" href="data_preprocessing.html#trim-adapters">trim_adapters</a></li>
<li class="toctree-l2"><a class="reference internal" href="data_preprocessing.html#remove-duplicates">remove_duplicates</a></li>
<li class="toctree-l2"><a class="reference internal" href="data_preprocessing.html#filter-contaminants">filter_contaminants</a></li>
<li class="toctree-l2"><a class="reference internal" href="data_preprocessing.html#visualise-quality-metrics">visualise_quality_metrics</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="statistical_analysis.html">Statistical Analysis</a><ul>
<li class="toctree-l2"><a class="reference internal" href="statistical_analysis.html#calculate-gc-content">calculate_gc_content</a></li>
<li class="toctree-l2"><a class="reference internal" href="statistical_analysis.html#count-nucleotides">count_nucleotides</a></li>
<li class="toctree-l2"><a class="reference internal" href="statistical_analysis.html#reverse-complement">reverse_complement</a></li>
<li class="toctree-l2"><a class="reference internal" href="statistical_analysis.html#translate">translate</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="alignment.html">Alignment</a><ul>
<li class="toctree-l2"><a class="reference internal" href="alignment.html#align">align</a></li>
<li class="toctree-l2"><a class="reference internal" href="alignment.html#read-alignment">read_alignment</a></li>
<li class="toctree-l2"><a class="reference internal" href="alignment.html#pairwise-alignment">pairwise_alignment</a></li>
<li class="toctree-l2"><a class="reference internal" href="alignment.html#merge-alignment">merge_alignment</a></li>
<li class="toctree-l2"><a class="reference internal" href="alignment.html#evaluate-alignment-quality">evaluate_alignment_quality</a></li>
<li class="toctree-l2"><a class="reference internal" href="alignment.html#extract-conserved-regions">extract_conserved_regions</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="annotation.html">Annotation</a><ul>
<li class="toctree-l2"><a class="reference internal" href="annotation.html#extract-annotations">extract_annotations</a></li>
<li class="toctree-l2"><a class="reference internal" href="annotation.html#filter-annotations">filter_annotations</a></li>
<li class="toctree-l2"><a class="reference internal" href="annotation.html#search-annotations">search_annotations</a></li>
<li class="toctree-l2"><a class="reference internal" href="annotation.html#extract-locations">extract_locations</a></li>
<li class="toctree-l2"><a class="reference internal" href="annotation.html#visualize-annotations">visualize_annotations</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="variant_analysis.html">variant analysis</a><ul>
<li class="toctree-l2"><a class="reference internal" href="variant_analysis.html#variant-calling">variant_calling</a></li>
<li class="toctree-l2"><a class="reference internal" href="variant_analysis.html#variant-annotation">variant_annotation</a></li>
<li class="toctree-l2"><a class="reference internal" href="variant_analysis.html#variant-effect-prediction">variant_effect_prediction</a></li>
<li class="toctree-l2"><a class="reference internal" href="variant_analysis.html#splice-site-prediction">splice_site_prediction</a></li>
<li class="toctree-l2"><a class="reference internal" href="variant_analysis.html#phenotype-prediction">phenotype_prediction</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="gene_expression.html">gene expression</a><ul>
<li class="toctree-l2"><a class="reference internal" href="gene_expression.html#read-gene-expression-file">read_gene_expression_file</a></li>
<li class="toctree-l2"><a class="reference internal" href="gene_expression.html#gene-expression-normalization">gene_expression_normalization</a></li>
<li class="toctree-l2"><a class="reference internal" href="gene_expression.html#get-group-labels">get_group_labels</a></li>
<li class="toctree-l2"><a class="reference internal" href="gene_expression.html#differential-gene-expression-analysis">differential_gene_expression_analysis</a></li>
<li class="toctree-l2"><a class="reference internal" href="gene_expression.html#gene-expression-quantification">gene_expression_quantification</a></li>
<li class="toctree-l2"><a class="reference internal" href="gene_expression.html#gene-expression-correlation">gene_expression_correlation</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="comparative_genomics.html">comparative genomics</a><ul>
<li class="toctree-l2"><a class="reference internal" href="comparative_genomics.html#compare-genome-size">compare_genome_size</a></li>
<li class="toctree-l2"><a class="reference internal" href="comparative_genomics.html#identify-conserved-regions">identify_conserved_regions</a></li>
<li class="toctree-l2"><a class="reference internal" href="comparative_genomics.html#pairwise-identity">pairwise_identity</a></li>
<li class="toctree-l2"><a class="reference internal" href="comparative_genomics.html#phylogenetic-distance">phylogenetic_distance</a></li>
</ul>
</li>
</ul>
</div>
</section>
</section>
<section id="indices-and-tables">
<h1>Indices and tables<a class="headerlink" href="#indices-and-tables" title="Permalink to this heading"></a></h1>
<ul class="simple">
<li><p><a class="reference internal" href="genindex.html"><span class="std std-ref">Index</span></a></p></li>
<li><p><a class="reference internal" href="py-modindex.html"><span class="std std-ref">Module Index</span></a></p></li>
<li><p><a class="reference internal" href="search.html"><span class="std std-ref">Search Page</span></a></p></li>
</ul>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="data_preprocessing.html" class="btn btn-neutral float-right" title="Data Preprocessing" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2023, Sayali Garud.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>