<!DOCTYPE html>

<html lang="python">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

    <title>variant analysis &#8212; GenBioX 0.3.1 documentation</title>
    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="_static/alabaster.css" />
    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/sphinx_highlight.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="gene expression" href="gene_expression.html" />
    <link rel="prev" title="Annotation" href="annotation.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <section id="variant-analysis">
<h1>variant analysis<a class="headerlink" href="#variant-analysis" title="Permalink to this heading">¶</a></h1>
<p>Prediction of the functional effects of genetic variants</p>
<section id="variant-calling">
<h2>variant_calling<a class="headerlink" href="#variant-calling" title="Permalink to this heading">¶</a></h2>
<p>Identifies genetic variants such as SNPs, indels, and structural variations from sequencing data.</p>
<p>Usage:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">GenBioX.variant_analysis</span> <span class="k">as</span> <span class="nn">va</span>

<span class="n">va</span><span class="o">.</span><span class="n">ariant_calling</span><span class="p">(</span><span class="n">input_bam</span><span class="p">,</span> <span class="n">reference_genome</span><span class="p">,</span> <span class="n">output_vcf</span><span class="p">)</span>
</pre></div>
</div>
</section>
<section id="variant-annotation">
<h2>variant_annotation<a class="headerlink" href="#variant-annotation" title="Permalink to this heading">¶</a></h2>
<p>Annotates genetic variants with information such as gene annotation, variant type, location, and allele frequency.</p>
<p>Usage:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">va</span><span class="o">.</span><span class="n">variant_annotation</span><span class="p">(</span><span class="n">input_vcf</span><span class="p">,</span> <span class="n">output_annovar_prefix</span><span class="p">,</span> <span class="n">database</span><span class="p">)</span>
</pre></div>
</div>
</section>
<section id="variant-effect-prediction">
<h2>variant_effect_prediction<a class="headerlink" href="#variant-effect-prediction" title="Permalink to this heading">¶</a></h2>
<p>Predicts the functional consequences of genetic variants</p>
<p>Usage:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">va</span><span class="o">.</span><span class="n">variant_effect_prediction</span><span class="p">(</span><span class="n">input_vcf</span><span class="p">,</span> <span class="n">output_vcf</span><span class="p">,</span> <span class="n">reference_genome</span><span class="p">)</span>
</pre></div>
</div>
</section>
<section id="splice-site-prediction">
<h2>splice_site_prediction<a class="headerlink" href="#splice-site-prediction" title="Permalink to this heading">¶</a></h2>
<p>predicts the effect of variants on splicing by analyzing splice site sequences and motifs</p>
<p>Usage:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">va</span><span class="o">.</span><span class="n">splice_site_prediction</span><span class="p">(</span><span class="n">input_vcf</span><span class="p">,</span> <span class="n">output_vcf</span><span class="p">,</span> <span class="n">reference_genome</span><span class="p">):</span>
</pre></div>
</div>
</section>
<section id="phenotype-prediction">
<h2>phenotype_prediction<a class="headerlink" href="#phenotype-prediction" title="Permalink to this heading">¶</a></h2>
<p>predicts the effect of variants on phenotypes</p>
<p>Usage:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">va</span><span class="o">.</span><span class="n">phenotype_prediction</span><span class="p">(</span><span class="n">variants</span><span class="p">,</span> <span class="n">phenotypes</span><span class="p">)</span>
</pre></div>
</div>
</section>
</section>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="index.html">GenBioX</a></h1>








<h3>Navigation</h3>
<p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="data_preprocessing.html">Data Preprocessing</a></li>
<li class="toctree-l1"><a class="reference internal" href="statistical_analysis.html">Statistical Analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="alignment.html">Alignment</a></li>
<li class="toctree-l1"><a class="reference internal" href="annotation.html">Annotation</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">variant analysis</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#variant-calling">variant_calling</a></li>
<li class="toctree-l2"><a class="reference internal" href="#variant-annotation">variant_annotation</a></li>
<li class="toctree-l2"><a class="reference internal" href="#variant-effect-prediction">variant_effect_prediction</a></li>
<li class="toctree-l2"><a class="reference internal" href="#splice-site-prediction">splice_site_prediction</a></li>
<li class="toctree-l2"><a class="reference internal" href="#phenotype-prediction">phenotype_prediction</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="gene_expression.html">gene expression</a></li>
<li class="toctree-l1"><a class="reference internal" href="comparative_genomics.html">comparative genomics</a></li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
      <li>Previous: <a href="annotation.html" title="previous chapter">Annotation</a></li>
      <li>Next: <a href="gene_expression.html" title="next chapter">gene expression</a></li>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>document.getElementById('searchbox').style.display = "block"</script>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2023, Sayali Garud.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 7.0.0</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.12</a>
      
      |
      <a href="_sources/variant_analysis.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>